### PR TITLE
Frontend and HDF5 back-end changes from the file_backend

### PR DESCRIPTION
**Pull request by jgrewe**

### CODE REVIEW

File: src/hdf5/FeatureHDF5.cpp [hunk 1]
```diff
@@ -85,20 +85,13 @@ void FeatureHDF5::data(const std::string &name_or_id) {
 
 shared_ptr<IDataArray> FeatureHDF5::data() const {
     shared_ptr<IDataArray> da;
-    bool error = false;
 
     if (group().hasGroup("data")) {
         Group other_group = group().openGroup("data", false);
         da = make_shared<DataArrayHDF5>(file(), block, other_group);
-        if (!block->hasDataArray(da->id())) error = true;
+        if (!block->hasDataArray(da->id()))
```
Comment (gicmo) [inline src/hdf5/FeatureHDF5.cpp]: Shouldn't we stay consistent and have:

```c++
if (!block->hasDataArray(da->id())) {
    throw std::runtime_error("FeatureHDF5::data: DataArray not found!");
}
```

File: src/Section.cpp [hunk 2]
```diff
@@ -47,7 +47,7 @@ Section::Section(shared_ptr<base::ISection> &&ptr)
 
 
 void Section::link(const Section &link) {
-    if (link == none) {
+    if (!link) {
         backend()->link(none);
     } else {
         backend()->link(link.id());
```
Comment (gicmo) [inline src/Section.cpp]: I would make the whole thing into `backend()->link(link ? link.id() : none);`

File: src/hdf5/FeatureHDF5.cpp (hunk 1 above)
Comment (jgrewe) [inline src/hdf5/FeatureHDF5.cpp]: do you mean the curly braces?

Comment (gicmo) [inline src/hdf5/FeatureHDF5.cpp]: Yeah. ;)

File: src/Section.cpp (hunk 2 above)
Comment (jgrewe) [inline src/Section.cpp]: no match for ternary ‘operator?:’ (operand types are ‘bool’, ‘std::string {aka std::basic_string<char>}’, and ‘const none_t {aka int boost::detail::none_helper::* const}’)

Comment (gicmo) [inline src/Section.cpp]: Oh! na, has to be the same type. Sure. Sorry.

**Comment by gicmo** [comment]
LGTM

**Pull request merged**